src/common/resize-watcher.js

```diff
@@ -0,0 +1,77 @@
+//@flow
+import {FakeEvent, FakeEventTarget} from '@playkit-js/playkit-js';
+
+class ResizeWatcher extends FakeEventTarget {
+  _observer: Object;
+  _config: Object;
+
+  constructor(config: Object) {
+    super();
+    this._config = config;
+    this._initObserver();
+  }
+
+  destroy() {
+    this._observer.disconnect();
+    this._config = {};
+  }
+
+  _initObserver() {
+    window.ResizeObserver ? this._createNativeObserver() : this._createIframeObserver();
+    const el = document.getElementById(this._config.targetId);
+    this._observer.observe(el);
+  }
+
+  _createNativeObserver() {
+    this._observer = new window.ResizeObserver(entries => {
+      entries.forEach(() => {
+        this._triggerResize();
+      });
+    });
+  }
+
+  _createIframeObserver() {
+    this._observer = new IFrameObserver(this._triggerResize.bind(this));
+  }
+
+  _triggerResize() {
+    this.dispatchEvent(new FakeEvent('resize'));
```

Comment:
Define a `RESIZE` enum inside the `CustomEventType` of the `playkit.js` and import it here.